linker/symbols.go

```diff
@@ -232,29 +230,29 @@ func (s *packageSymbols) importPackage(pkgSpan ast.SourceSpan, pkg protoreflect.
 	return child, nil
 }
 
-func (s *Symbols) getPackage(pkg protoreflect.FullName) *packageSymbols {
+func (s *Symbols) getPackage(pkg protoreflect.FullName, exact bool) *packageSymbols {
 	if pkg == "" {
 		return &s.pkgTrie
 	}
-
-	parts := strings.Split(string(pkg), ".")
-	for i := 1; i < len(parts); i++ {
-		parts[i] = parts[i-1] + "." + parts[i]
-	}
-
 	cur := &s.pkgTrie
-	for _, p := range parts {
+	enumerator := nameEnumerator{name: pkg}
+	for {
+		p, ok := enumerator.next()
+		if !ok {
+			return cur
+		}
 		cur.mu.RLock()
-		next := cur.children[protoreflect.FullName(p)]
+		next := cur.children[p]
```

Comment:
Is it possible to miss a child if an package path has gaps, e.g. given `a.Type`, `a.b.c.Type` where `a` and `a.b.c` are packages would this report `a` as the package for both?